c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -494,7 +494,80 @@ impl<'tcx> TypeChecker<'tcx, '_> {
                 let perms = PermissionSet::FREE;
                 self.constraints.add_all_perms(rv_lty.label, perms);
             }
+            Callee::Memcpy => {
```

Comment:
At some point we should extend this to include a pseudo-assignment `*dest = *src;`.  We expect to produce safe rewrites for at least some uses of memcpy, and in those cases we should make sure `dataflow` treats it like the safe version (particularly in cases where `*src` is a pointer or contains one).

Reply:
`const` vs `mut` maybe matters for `dest = src`, but it shouldn't matter for `*dest = *src`, which is effectively what's happening in `memcpy` (at least when the number of bytes to copy is equal to `size_of::<T>()`).

Reply:
Yes, that seems about right